xhype/xhype/src/utils.rs

```diff
@@ -0,0 +1,10 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#[inline]
+pub fn round_up(num: usize) -> usize {
```

Comment:
I updated.

Reply:
Your addition could overflow, which would panic!(), for example for an address at the top of the virtual address space. I'd recommend using `wrapping_add` instead of `+` to permit this.

Reply:
Thanks for the suggestion! but I do not think panic!() on arithmetic overflow is a bad thing here since the wrong result makes no sense and the program should stop. However, as far as I know, in release mode Rust will not check arithmetic overflow. Therefore I think I should manually check overflow and if so, call panic!().

Reply:
It may be ok, depending on what's at the top of the address space. For example, if the highest page in the system contained a stack, then overflow would in fact be correct (you'd load 0 into %rsp, and since the stack is full/descending on x86_64, you'd get the correct behavior).

It is true that in release mode the checks are optimized away; regardless, there's no point not to use the wrapping_add if that's the behavior you want.

You _can_ selectively enable overflow checking regardless of compilation mode, however, if you want to trap on overflow: `-Z force-overflow-checks` of `#[overflow_checks(on)] `.